### PR TITLE
Retry failed outgoing chat messages

### LINKED ISSUE #377: ChatViewModel retries failed outgoing messages
## Problem statement
Part of #213.

Failed outgoing messages must be retryable through the ChatViewModel action model.

## Proposed solution
Add a ViewModel retry action for failed outgoing messages.

Retry should:
- target a failed outgoing message by MessageId
- reuse the same message id and text
- transition the outgoing message back to Sending
- update it to Sent or Failed based on the retry result

## Acceptance criteria
- [ ] ViewModel exposes an action to retry a failed outgoing message.
- [ ] Retry reuses the failed message id.
- [ ] Retry reuses the failed message text.
- [ ] Retry success updates state and clears input only after success when applicable.
- [ ] Retry failure keeps the message failed and retryable.
- [ ] Tests cover retry success and retry failure.

### PR DESCRIPTION
**Pull request by timurgilfanov**
Closes #377

## Summary
Adds retry support for failed outgoing chat messages through the `ChatViewModel` action model. Failed outgoing messages can be retried by `MessageId`, preserving the original id and text while transitioning back through sending state and then to sent or failed based on the retry result.

Retry preserves the original failed outgoing message identity instead of creating a new message. That keeps UI state stable and matches the backend-facing contract that a retry represents another send attempt for the same outgoing message.

## Changes
- Added `ChatViewModel` retry handling for failed outgoing messages.
- Updated send-message domain behavior to allow retrying with an existing message id.
- Added focused retry success and failure coverage for chat ViewModel behavior.
- Added domain test coverage for message creation with a supplied id.
- Documented the implementation plan for retrying failed outgoing messages.

## Important Design Decision
None.

## Testing
- [x] Unit tests added or updated
- [ ] Manually tested
- [x] Edge cases considered

## Screenshots / Demo
N/A. This change updates chat state handling and tests, with no visual UI changes.

## Acceptance criteria
- [x] ViewModel exposes an action to retry a failed outgoing message.
- [x] Retry reuses the failed message id.
- [x] Retry reuses the failed message text.
- [x] Retry success updates state and clears input only after success when applicable.
- [x] Retry failure keeps the message failed and retryable.
- [x] Tests cover retry success and retry failure.

## Breaking Changes
None.

## Follow-ups
None.

### CODE REVIEW

File: app/src/main/java/timur/gilfanov/messenger/ui/screen/chat/ChatViewModel.kt [hunk 1]
```diff
@@ -198,15 +208,53 @@ class ChatViewModel @AssistedInject constructor(
...
 
-    private fun textMessage(messageId: MessageId, now: Instant, text: String): TextMessage =
-        TextMessage(
-            id = messageId,
-            parentId = null,
-            sender = currentChat!!.participants.first { it.isCurrentUser },
-            recipient = chatId,
-            createdAt = now,
-            text = text,
-        )
```
Comment (timurgilfanov) [inline app/src/main/java/timur/gilfanov/messenger/ui/screen/chat/ChatViewModel.kt:209]: Why this is removed?

File: app/src/main/java/timur/gilfanov/messenger/ui/screen/chat/ChatViewModel.kt [hunk 2]
```diff
@@ -224,14 +272,16 @@ class ChatViewModel @AssistedInject constructor(
...
+                if (result is ResultWithError.Success) {
+                    currentChat = result.data
+                }
+            }
             .debounce(STATE_UPDATE_DEBOUNCE)
             .collect { result ->
                 when (result) {
-                    is ResultWithError.Success -> {
-                        val chat = result.data
-                        currentChat = chat
```
Comment (timurgilfanov) [inline app/src/main/java/timur/gilfanov/messenger/ui/screen/chat/ChatViewModel.kt:232]: Why `currentChat` update moved to onEach?

File: app/src/test/java/timur/gilfanov/messenger/ui/screen/chat/ChatViewModelRetryTest.kt [hunk 3]
```diff
@@ -0,0 +1,485 @@
...
+        assertEquals(1, repository.sentMessages.size)
+        repository.assertRetriedMessageIsFailedOne()
+        val afterFailure = viewModel.state.value
+        assertTrue(afterFailure is ChatUiState.Ready)
+        val dialogError = assertIs<ReadyError.SendMessageError>(afterFailure.dialogError)
+        assertIs<SendMessageError.DeliveryStatusUpdateNotValid>(dialogError.error)
+    }
+
+    @Test
+    fun `retry succeeds under debounce rule despite its own failed timeline entry`() = runTest {
```
Comment (timurgilfanov) [inline app/src/test/java/timur/gilfanov/messenger/ui/screen/chat/ChatViewModelRetryTest.kt]: Does this test duplicate test `debounce excludes the message being re-sent from its own baseline` in `CreateMessageUseCaseTest.kt`?

File: app/src/main/java/timur/gilfanov/messenger/ui/screen/chat/ChatViewModel.kt (hunk 1 above)
Comment (timurgilfanov) [reply app/src/main/java/timur/gilfanov/messenger/ui/screen/chat/ChatViewModel.kt:209]: The \`textMessage(messageId, now, text)\` helper had only one caller — \`sendMessage\`. The new \`retryMessage\` path does not build a fresh message; it reuses the existing failed timeline entry via \`failed.copy(deliveryStatus = null)\`. So the helper stayed single-use and was inlined back into \`sendMessage\` so the \`TextMessage\` construction sits next to where it is sent. No behavior change — same fields. Happy to restore it as a named helper if you prefer, but \`retryMessage\` would not share it.

File: app/src/main/java/timur/gilfanov/messenger/ui/screen/chat/ChatViewModel.kt (hunk 2 above)
Comment (timurgilfanov) [reply app/src/main/java/timur/gilfanov/messenger/ui/screen/chat/ChatViewModel.kt:232]: \`currentChat\` is the domain-facing snapshot that \`sendMessage\`/\`retryMessage\` read synchronously (to find the failed message, resolve the current-user sender, and pass the chat into the use case), whereas \`_state\` is intentionally debounced. If \`currentChat\` were assigned after \`.debounce(STATE_UPDATE_DEBOUNCE)\` (as before), it would lag the latest received chat by up to the debounce window, so a retry/send firing in that window would operate on a stale chat (e.g. not yet seeing a message that just flipped to Failed). Moving the assignment into \`.onEach\` (before \`.debounce\`) keeps \`currentChat\` current for domain operations while the UI still updates with debouncing. The two concerns are now decoupled.

File: app/src/test/java/timur/gilfanov/messenger/ui/screen/chat/ChatViewModelRetryTest.kt (hunk 3 above)
Comment (timurgilfanov) [reply app/src/test/java/timur/gilfanov/messenger/ui/screen/chat/ChatViewModelRetryTest.kt]: Yes — you are right. After checking the wiring (\`createViewModel\` uses the real \`SendMessageUseCase\`, whose \`checkDebounceRule\` excludes the message via \`lastMessageBy(..., excluding = message.id)\`), this test was the intersection of two existing ones: ViewModel id-reuse + reaching Sent is covered by \`retry re-sends same id and text, clears status, and reaches Sent\`, and the debounce-baseline exclusion is covered by \`debounce excludes the message being re-sent from its own baseline\` in CreateMessageUseCaseTest. The ViewModel forwards chat.rules straight to the real use case, leaving no untested seam. Removed it in c5138290.
